server.js

```diff
@@ -1,38 +1,183 @@
+  process.env.MONGO_URL || 'mongodb://localhost/skotrum-mongo-project';
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const Skotrum = mongoose.model('Skotrum', {
+  id: Number,
+  name: String,
+  adress: String,
+  phone: String,
+  openHours: String,
+  note: String,
+  webpage: String,
+  location: String,
+  hasChangingRooms: Boolean
+});
+
+const Restaurant = mongoose.model('Restaurant', {
+  name: String,
+  phone: String,
+  webpage: String,
+  adress: {
+    type: mongoose.Schema.Types.String,
+    ref: 'BabyRooms'
+  }
+});
+
+const Location = mongoose.model('Location', {
+  name: String,
+  adress: String,
+  location: {
+    type: mongoose.Schema.Types.String,
+    ref: 'Restaurant'
+  }
+});
+
+const OpenHours = mongoose.model('OpenHours', {
+  name: String,
+  openHours: {
+    type: mongoose.Schema.Types.String,
+    ref: 'Location'
+  }
+});
+
+const BabyRooms = mongoose.model('BabyRooms', {
+  name: String,
+  note: String,
+  hasChangingRooms: {
+    type: mongoose.Schema.Types.Boolean,
+    ref: 'Restaurant'
+  }
+});
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Skotrum.deleteMany();
+    await Restaurant.deleteMany();
+    await Location.deleteMany();
+    await OpenHours.deleteMany();
+    await BabyRooms.deleteMany();
+
+    skotrum.forEach(restData => {
+      new Skotrum(restData).save();
+      new Restaurant(restData).save();
+      new Location(restData).save();
+      new OpenHours(restData).save();
+      new BabyRooms(restData).save();
+    });
+  };
+  seedDatabase();
+}
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+// Middlewares to enable cors and json body parsing
+app.use(cors());
+app.use(bodyParser.json());
 
-// Start defining your routes here
+// Routes
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Skotrum');
+});
+
+//to find all babyrooms in sthlm
+app.get('/findbabyrooms', async (req, res) => {
+  const skotrum = await Skotrum.find();
+  console.log(skotrum);
+
+  res.json(skotrum);
+});
+
+//to find one babyroom by id
+app.get('/findbabyrooms/:id', async (req, res) => {
+  const skotrum = await Skotrum.findById(req.params.id);
+  if (skotrum) {
+    res.json(skotrum);
+  } else {
+    res.status(404).json({ error: 'restaurant not found' });
+  }
+});
+
+// to find restaurants with babyrooms including name and adress
+app.get('/contacts', async (req, res) => {
+  const contact = await Restaurant.find().populate('Location');
+  if (contact) {
+    res.json(contact);
+  } else {
+    res.status(404).json({ error: 'restauarnt not found' });
```

Comment:
This is an example of when it can be handy to define your error message
`const ERROR_RESTURANT_NOT_FOUND = 'Restaurant not found'`

Reply:
ok, thanks